feat: `keypair_name` and `last_used_derivation_index` columns added to `accounts` table

saledjenic opened the pull request:
- `keypair_name` added to `accounts` table, all accounts derived from the same master key have the same keypair name and also no two keypairs share the same keypair name (keypair name is unique per keypair)
- `last_used_derivation_index` added to `accounts` table, cause we need to maintain the highest index been used for the derivations made within the same keypair

@status-im/mobile-devs I am not sure if you're using `latest_derived_path` from the `settings` table or not (the only thing why the desktop app used it for was to properly create/derive next wallet account), but if not I would remove that column form the `settings` table within this PR?

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (99)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | faaabf5d | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/1/) | 2023-03-08 15:27:16 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230308-152551-faaabf-pr3268.zip) |
| :heavy_multiplication_x: | faaabf5d | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/1/) | 2023-03-08 15:28:32 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/1//consoleText) |
| :heavy_check_mark: | faaabf5d | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/1/) | 2023-03-08 15:30:24 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230308-152551-faaabf-pr3268.zip) |
| :heavy_check_mark: | faaabf5d | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/1/) | 2023-03-08 15:30:57 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230308-152551-faaabf-pr3268.aar) |
| | | | | | | |
| :heavy_check_mark: | 6dd11983 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/2/) | 2023-03-08 19:05:32 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230308-190412-6dd119-pr3268.zip) |
| :heavy_check_mark: | 6dd11983 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/2/) | 2023-03-08 19:07:17 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230308-190407-6dd119-pr3268.zip) |
| :heavy_check_mark: | 6dd11983 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/2/) | 2023-03-08 19:08:00 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230308-190407-6dd119-pr3268.aar) |
| :heavy_multiplication_x: | 6dd11983 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/2/) | 2023-03-08 19:11:57 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/2//consoleText) |
| :heavy_multiplication_x: | 6dd11983 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/3/) | 2023-03-08 19:57:06 | ~20 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e8f3f453 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/3/) | 2023-03-10 07:36:42 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230310-073517-e8f3f4-pr3268.zip) |
| :heavy_check_mark: | e8f3f453 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/3/) | 2023-03-10 07:39:04 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230310-073512-e8f3f4-pr3268.aar) |
| :heavy_check_mark: | e8f3f453 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/3/) | 2023-03-10 07:45:23 | ~10 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230310-073517-e8f3f4-pr3268.zip) |
| :heavy_multiplication_x: | e8f3f453 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/4/) | 2023-03-10 07:56:58 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/4//consoleText) |
| | | | | | | |
| :heavy_check_mark: | de37075b | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/4/) | 2023-03-10 07:38:45 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230310-073742-de3707-pr3268.zip) |
| :heavy_check_mark: | de37075b | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/4/) | 2023-03-10 07:42:36 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230310-073909-de3707-pr3268.aar) |
| :heavy_check_mark: | de37075b | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/4/) | 2023-03-10 07:48:14 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230310-074526-de3707-pr3268.zip) |
| :heavy_multiplication_x: | de37075b | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/5/) | 2023-03-10 08:18:58 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/5//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7e4ec46a | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/5/) | 2023-03-13 17:10:29 | ~59 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230313-170929-7e4ec4-pr3268.zip) |
| :heavy_check_mark: | 7e4ec46a | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/5/) | 2023-03-13 17:12:20 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230313-170929-7e4ec4-pr3268.zip) |
| :heavy_check_mark: | 7e4ec46a | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/5/) | 2023-03-13 17:12:56 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230313-170929-7e4ec4-pr3268.aar) |
| :heavy_multiplication_x: | 7e4ec46a | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/6/) | 2023-03-13 17:31:02 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/6//consoleText) |
| | | | | | | |
| :heavy_check_mark: | c541ae00 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/6/) | 2023-03-15 09:59:44 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230315-095841-c541ae-pr3268.zip) |
| :heavy_check_mark: | c541ae00 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/6/) | 2023-03-15 10:01:35 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230315-095837-c541ae-pr3268.zip) |
| :heavy_check_mark: | c541ae00 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/6/) | 2023-03-15 10:02:17 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230315-095837-c541ae-pr3268.aar) |
| :heavy_multiplication_x: | c541ae00 | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/7/) | 2023-03-15 10:20:32 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | e7513927 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/7/) | 2023-03-15 10:03:52 | ~55 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230315-100256-e75139-pr3268.zip) |
| :heavy_check_mark: | e7513927 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/7/) | 2023-03-15 10:05:20 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230315-100252-e75139-pr3268.zip) |
| :heavy_check_mark: | e7513927 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/7/) | 2023-03-15 10:06:27 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230315-100252-e75139-pr3268.aar) |
| :heavy_multiplication_x: | e7513927 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/8/) | 2023-03-15 10:42:40 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bdbdb772 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/8/) | 2023-03-16 19:14:47 | ~58 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230316-191348-bdbdb7-pr3268.zip) |
| :heavy_check_mark: | bdbdb772 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/8/) | 2023-03-16 19:16:21 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230316-191348-bdbdb7-pr3268.zip) |
| :heavy_check_mark: | bdbdb772 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/8/) | 2023-03-16 19:18:22 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230316-191348-bdbdb7-pr3268.aar) |
| :heavy_multiplication_x: | bdbdb772 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/9/) | 2023-03-16 19:36:31 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/9//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8e65c976 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/9/) | 2023-03-20 07:41:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230320-074007-8e65c9-pr3268.zip) |
| :heavy_check_mark: | 8e65c976 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/9/) | 2023-03-20 07:42:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230320-074007-8e65c9-pr3268.zip) |
| :heavy_check_mark: | 8e65c976 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/9/) | 2023-03-20 07:44:36 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230320-074007-8e65c9-pr3268.aar) |
| :heavy_multiplication_x: | 8e65c976 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/10/) | 2023-03-20 08:03:20 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/10//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 5027f101 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/10/) | 2023-03-20 11:28:24 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230320-112706-5027f1-pr3268.zip) |
| :heavy_check_mark: | 5027f101 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/10/) | 2023-03-20 11:29:37 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230320-112706-5027f1-pr3268.zip) |
| :heavy_check_mark: | 5027f101 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/10/) | 2023-03-20 11:30:21 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230320-112704-5027f1-pr3268.aar) |
| :heavy_multiplication_x: | 5027f101 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/11/) | 2023-03-20 11:48:27 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/11//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0d176d04 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/11/) | 2023-03-21 08:31:21 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230321-083002-0d176d-pr3268.zip) |
| :heavy_check_mark: | 0d176d04 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/11/) | 2023-03-21 08:32:58 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230321-083002-0d176d-pr3268.zip) |
| :heavy_check_mark: | 0d176d04 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/11/) | 2023-03-21 08:34:17 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230321-083002-0d176d-pr3268.aar) |
| :heavy_multiplication_x: | 0d176d04 | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/12/) | 2023-03-21 08:53:12 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/12//consoleText) |
| | | | | | | |
| :heavy_check_mark: | bf0ae262 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/12/) | 2023-03-21 12:18:55 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230321-121614-bf0ae2-pr3268.zip) |
| :heavy_check_mark: | bf0ae262 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/12/) | 2023-03-21 12:19:42 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230321-121614-bf0ae2-pr3268.zip) |
| :heavy_check_mark: | bf0ae262 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/12/) | 2023-03-21 12:24:07 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230321-121614-bf0ae2-pr3268.aar) |
| :heavy_multiplication_x: | bf0ae262 | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/13/) | 2023-03-21 12:42:41 | ~26 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/13//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 01d7c4ab | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/13/) | 2023-03-22 10:37:29 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230322-103451-01d7c4-pr3268.zip) |
| :heavy_check_mark: | 01d7c4ab | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/13/) | 2023-03-22 10:38:07 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230322-103455-01d7c4-pr3268.zip) |
| :heavy_multiplication_x: | 01d7c4ab | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/14/) | 2023-03-22 10:39:08 | ~4 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/14//consoleText) |
| :heavy_check_mark: | 01d7c4ab | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/13/) | 2023-03-22 10:39:15 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230322-103451-01d7c4-pr3268.aar) |
| | | | | | | |
| :heavy_check_mark: | b17ba7a9 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/14/) | 2023-03-22 16:08:46 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230322-160744-b17ba7-pr3268.zip) |
| :heavy_check_mark: | b17ba7a9 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/14/) | 2023-03-22 16:10:20 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230322-160744-b17ba7-pr3268.zip) |
| :heavy_check_mark: | b17ba7a9 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/14/) | 2023-03-22 16:11:09 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230322-160740-b17ba7-pr3268.aar) |
| :heavy_multiplication_x: | b17ba7a9 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/15/) | 2023-03-22 16:29:44 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/15//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 8e3975ab | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/15/) | 2023-03-23 15:04:16 | ~58 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230323-150317-8e3975-pr3268.zip) |
| :heavy_check_mark: | 8e3975ab | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/15/) | 2023-03-23 15:05:52 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230323-150317-8e3975-pr3268.zip) |
| :heavy_check_mark: | 8e3975ab | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/15/) | 2023-03-23 15:07:06 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230323-150312-8e3975-pr3268.aar) |
| :heavy_multiplication_x: | 8e3975ab | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/16/) | 2023-03-23 15:25:20 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/16//consoleText) |
| :heavy_multiplication_x: | 8e3975ab | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/17/) | 2023-03-23 15:34:09 | ~53 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/17//consoleText) |
| | | | | | | |
| :heavy_check_mark: | dee575d9 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/16/) | 2023-03-24 07:33:20 | ~59 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230324-073220-dee575-pr3268.zip) |
| :heavy_check_mark: | dee575d9 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/16/) | 2023-03-24 07:35:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230324-073220-dee575-pr3268.zip) |
| :heavy_check_mark: | dee575d9 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/16/) | 2023-03-24 07:35:52 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230324-073220-dee575-pr3268.aar) |
| :heavy_multiplication_x: | dee575d9 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/18/) | 2023-03-24 07:54:26 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/18//consoleText) |
| :heavy_multiplication_x: | dee575d9 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/19/) | 2023-03-24 09:23:27 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/19//consoleText) |
| :heavy_multiplication_x: | dee575d9 | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/20/) | 2023-03-24 10:07:55 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/20//consoleText) |
| :heavy_multiplication_x: | dee575d9 | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/21/) | 2023-03-24 10:32:20 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/21//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 93b46888 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/17/) | 2023-03-24 19:02:53 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230324-190106-93b468-pr3268.zip) |
| :heavy_check_mark: | 93b46888 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/17/) | 2023-03-24 19:04:30 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230324-190106-93b468-pr3268.zip) |
| :heavy_check_mark: | 93b46888 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/17/) | 2023-03-24 19:04:44 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230324-190106-93b468-pr3268.aar) |
| :heavy_multiplication_x: | 93b46888 | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/22/) | 2023-03-24 19:23:34 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/22//consoleText) |
| :heavy_multiplication_x: | 93b46888 | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/23/) | 2023-03-24 19:27:07 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/23//consoleText) |
| :heavy_multiplication_x: | 93b46888 | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/24/) | 2023-03-24 19:28:39 | ~52 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/24//consoleText) |
| :heavy_multiplication_x: | 93b46888 | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/25/) | 2023-03-24 19:56:33 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/25//consoleText) |
| :heavy_multiplication_x: | 93b46888 | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/26/) | 2023-03-24 20:18:28 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/26//consoleText) |
| :heavy_multiplication_x: | 93b46888 | [#27](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/27/) | 2023-03-27 07:24:50 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/27//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 28f177ec | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/18/) | 2023-03-27 13:51:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230327-134818-28f177-pr3268.zip) |
| :heavy_check_mark: | 28f177ec | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/18/) | 2023-03-27 13:52:44 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230327-134818-28f177-pr3268.zip) |
| :heavy_check_mark: | 28f177ec | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/18/) | 2023-03-27 13:52:49 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230327-134814-28f177-pr3268.aar) |
| :heavy_multiplication_x: | 28f177ec | [#28](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/28/) | 2023-03-27 14:11:44 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/28//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 323317e4 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/19/) | 2023-03-27 14:37:31 | ~56 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230327-143634-323317-pr3268.zip) |
| :heavy_multiplication_x: | 323317e4 | [#29](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/29/) | 2023-03-27 14:37:46 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/29//consoleText) |
| :heavy_check_mark: | 323317e4 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/19/) | 2023-03-27 14:40:58 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230327-143634-323317-pr3268.aar) |
| :heavy_check_mark: | 323317e4 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/19/) | 2023-03-27 14:45:58 | ~9 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230327-143634-323317-pr3268.zip) |
| :heavy_multiplication_x: | 323317e4 | [#30](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/30/) | 2023-03-27 15:03:59 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/30//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 70953bba | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/20/) | 2023-03-27 15:52:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230327-155100-70953b-pr3268.zip) |
| :heavy_check_mark: | 70953bba | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/20/) | 2023-03-27 15:54:34 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230327-155100-70953b-pr3268.aar) |
| :heavy_check_mark: | 70953bba | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/20/) | 2023-03-27 16:00:28 | ~9 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230327-155100-70953b-pr3268.zip) |
| :heavy_multiplication_x: | 70953bba | [#31](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/31/) | 2023-03-27 16:12:50 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/31//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2809d78d | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/21/) | 2023-03-28 06:47:19 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230328-064614-2809d7-pr3268.zip) |
| :heavy_check_mark: | 2809d78d | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/21/) | 2023-03-28 06:49:51 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230328-064614-2809d7-pr3268.aar) |
| :heavy_check_mark: | 2809d78d | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/21/) | 2023-03-28 06:55:30 | ~9 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230328-064614-2809d7-pr3268.zip) |
| :heavy_multiplication_x: | 2809d78d | [#32](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/32/) | 2023-03-28 07:08:20 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/32//consoleText) |
| | | | | | | |
| :heavy_check_mark: | da73a4fb | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/22/) | 2023-03-28 12:02:24 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230328-120121-da73a4-pr3268.zip) |
| :heavy_check_mark: | da73a4fb | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/22/) | 2023-03-28 12:04:46 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230328-120121-da73a4-pr3268.aar) |
| :heavy_check_mark: | da73a4fb | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/22/) | 2023-03-28 12:10:35 | ~9 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230328-120121-da73a4-pr3268.zip) |
| :heavy_multiplication_x: | da73a4fb | [#33](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/33/) | 2023-03-28 12:22:54 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/33//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b2bbcceb | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/23/) | 2023-03-28 12:50:14 | ~58 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230328-124916-b2bbcc-pr3268.zip) |
| :heavy_multiplication_x: | b2bbcceb | [#34](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/34/) | 2023-03-28 12:50:30 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/34//consoleText) |
| :heavy_check_mark: | b2bbcceb | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/23/) | 2023-03-28 12:52:58 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230328-124911-b2bbcc-pr3268.aar) |
| :heavy_check_mark: | b2bbcceb | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/23/) | 2023-03-28 13:04:50 | ~15 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230328-124911-b2bbcc-pr3268.zip) |
| :heavy_multiplication_x: | b2bbcceb | [#35](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/35/) | 2023-03-28 13:12:53 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/35//consoleText) |
| :heavy_multiplication_x: | b2bbcceb | [#36](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/36/) | 2023-03-28 13:45:08 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/36//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 099666dd | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3268/24/) | 2023-03-28 14:04:45 | ~58 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230328-140346-099666-pr3268.zip) |
| :heavy_check_mark: | 099666dd | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3268/24/) | 2023-03-28 14:07:45 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230328-140342-099666-pr3268.aar) |
| :heavy_check_mark: | 099666dd | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/24/) | 2023-03-28 14:13:06 | ~9 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230328-140346-099666-pr3268.zip) |
| :heavy_check_mark: | 099666dd | [#37](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/37/) | 2023-03-28 14:13:48 | ~10 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3268/37//consoleText) |

ilmotta commented:
> 

Hey @saledjenic, I searched the mobile codebase, but I couldn't find any reference to `latest_derived_path`. It should be fine to remove the column, but maybe somebody else has a concern.

qfrank commented:
> I am not sure if you're using latest_derived_path from the settings table  

Hi @saledjenic , mobile used `latest_derived_path`

saledjenic committed:
feat: `keypair_name` and `last_used_derivation_index` columns added to `accounts` table

- `keypair_name` added to `accounts` table, all accounts derived from the
same master key have the same keypair name and also no two keypairs share
the same keypair name (keypair name is unique per keypair)
- `last_used_derivation_index` added to `accounts` table, cause we need
to maintain the highest index been used for the derivations made within
the same keypair

saledjenic committed:
feat: create acc from private key endpoint added

`CreateAccountFromPrivateKey` function added, it generates an account based on provided
private key, but doesn't store anything anywhere.

saledjenic committed:
neat: accounts and wallet api sorted out

Unused endpoints removed, new ones with more meaningful naming are added and
their purposes were revised.

saledjenic commented:
@richard-ramos @qfrank @alaibe please review this one again cause I added second and third commit as part of this PR, cause I need those additional changes for my "add wallet account" desktop related work.

- first commit: `keypair_name and last_used_derivation_index columns added to accounts table`, this one was already reviewed
- second commit `create acc from private key endpoint added` a new function `CreateAccountFromPrivateKey` was added, which creates an account from the provided private key without creating keystore files and adding anything to db.
- third commit ` accounts and wallet api sorted out` api endpoints are sorted out a bit, those unused are removed some new are added according to the desktop app needs.
  - **removed endpoints (accounts api)**:
    - `AddAccountWatch`
    - `AddAccountWithMnemonic`
    - `AddAccountWithMnemonicPasswordVerified`
    - `AddAccountWithMnemonicAndPath`
    - `AddAccountWithMnemonicAndPathPasswordVerified`
    - `AddAccountWithPrivateKeyPasswordVerified`
    - `AddAccountWithPrivateKey`
    - `GenerateAccount`
    - `GenerateAccountPasswordVerified`
    - `GenerateAccountWithDerivedPath`
  - **newly added endpoints (accounts api)**:
    - `AddAccount`
    - `ImportPrivateKey`
    - `ImportMnemonic`
    - `GetRandomMnemonic`
  - **removed endpoints (wallet api)**:
    - `GetDerivedAddressForPath`
    - `GetDerivedAddressesForPath`
    - `GetDerivedAddressesForMnemonicWithPath`
    - `GetDerivedAddressForPrivateKey`
    - `GetDerivedAddressDetails`
  - **newly added endpoints (wallet api)**:
    - `GetDerivedAddresses`
    - `GetDerivedAddressesForMnemonic`
    - `GetAddressDetails`

alaibe commented on multiaccounts/accounts/database.go:
```diff
@@ -129,6 +144,112 @@ func (db *Database) Close() error {
+	return err
+}
+
+func updateAccountsTableIfNeeded(db *sql.DB) error {
```
nit: this function name could probably have a better name e.g what to update and what is needed?

saledjenic replied:
fully agree, will update it later

saledjenic replied:
done

saledjenic committed:
feat: `GetRandomMnemonic` endpoint added to accounts api

saledjenic committed:
tests: updated due to updating old accounts as a part of migration process

status-im-auto commented:
#### :heavy_check_mark: [status-go/prs/ios/PR-3268#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3268/19/) :small_blue_diamond: ~9 min 35 sec :small_blue_diamond: 323317e4 :small_blue_diamond: [:package: ios package](https://status-im-prs.ams3.digitaloceanspaces.com/status-go-ios-230327-143634-323317-pr3268.zip)
